packages/terms-of-service/tests/php/test_TermsOfService.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+
+namespace Automattic\Jetpack;
+
+use PHPUnit\Framework\TestCase;
+use phpmock\Mock;
+use phpmock\MockBuilder;
+
+class Test_Terms_Of_Service extends TestCase {
+
+	/**
+	 * Test setup.
+	 */
+	public function setUp() {
+		$this->terms_of_service = $this->createPartialMock( __NAMESPACE__ .'\\Terms_Of_Service',
+			array( 'get_raw_has_agreed', 'is_development_mode', 'is_active', 'set_agree', 'set_revoke' )
+		);
+	}
+
+	/**
+	 * Test teardown.
+	 */
+	public function tearDown() {
+		Mock::disableAll();
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Terms_Of_Service->agree
+	 */
+	public function test_agree() {
+		$this->mock_function( 'do_action', null );
+		$this->terms_of_service->expects( $this->once() )->method( 'set_agree' )->willReturn( null );
+
+		$this->terms_of_service->agree();
+	}
+
+	/**
+	 * @covers Automattic\Jetpack\Terms_Of_Service->revoke
+	 */
+	public function test_revoke() {
```

Comment:
In this test, how can we verify that after the `->revoke` method has ran, the master user has declined agreement to the ToS?

Reply:
`expects( $this->once() )->method( 'set_revoke' )` does that in the test.

Reply:
Well, I expect it should have a last step that I don't see here:

* We mock whatever we want/need to mock.
* We call `->revoke()`.
* We test if it's been revoked.

Perhaps I misunderstand the purpose of the test?